packages/freezed/lib/src/templates/concrete_template.dart

```diff
@@ -154,12 +156,32 @@ ${copyWith.abstractCopyWithGetter}
 
   String get _redirectedFromJsonConstructor {
     if (!shouldGenerateJson) return '';
-    return 'factory ${constructor.redirectedName}.fromJson(Map<String, dynamic> json) = $concreteName$genericsParameter.fromJson;';
+
+    final genericArgs = hasGenericArgumentFactories
+        ? genericsParameter.typeParameters.map((type) {
+            return ', $type Function(Object? json) fromJson$type';
+          }).join()
+        : '';
+
+    return 'factory ${constructor.redirectedName}.fromJson(Map<String, dynamic> json$genericArgs) = $concreteName$genericsParameter.fromJson;';
   }
 
   String get _concreteFromJsonConstructor {
     if (!shouldGenerateJson) return '';
-    return 'factory $concreteName.fromJson(Map<String, dynamic> json) => _\$${concreteName}FromJson(json);';
+
+    final genericArgs = hasGenericArgumentFactories
+        ? genericsParameter.typeParameters.map((type) {
+            return ', $type Function(Object? json) fromJson$type';
+          }).join()
+        : '';
```

Comment:
to extract in a function to reuse with _redirectedFromJsonConstructor

Reply:
I've extracted it into a new private getter